natlib/svcd.c

```diff
@@ -30,6 +31,63 @@ static const LUA_REG_TYPE svcd_meta_map[] =
 // Maintainer: Michael Andersen <michael@steelcode.com>
 /////////////////////////////////////////////////////////////
 
+//Lua: storm.n.svcd_advert_received(pay, srcip, srcport)
+//Dispatch an advertisment
+
+int svcd_advert_received(lua_State *L) 
+{
+	char *srcip = luaL_checkstring(L, 2);
+        printf("Service advertisment %s\n", srcip);
+	
+	//check if pay is nil, don't proceed if true
+	int nil_pay=lua_isnil(L,1);
+        if(nil_pay==1)
+	{
+ 		return 0;
+	}
+
```

Comment:
You should consider just doing
<code><pre>if (lua_isnil(L, 1))
{
    return 0;
}</pre></code>

instead of explicitly checking if nil_pay is 1.